src/main/java/omero/cmd/graphs/BaseGraphTraversalProcessor.java

```diff
@@ -58,6 +75,23 @@ public void deleteInstances(String className, Collection<Long> ids) throws Graph
         if (count != ids.size()) {
             LOGGER.warn("not all the objects of type " + className + " could be deleted");
         }
+        if (DELETION_CLASS_NAMES.contains(className)) {
+            deletionLog.add(new AbstractMap.SimpleImmutableEntry<>(className.substring(className.lastIndexOf('.') + 1), ids));
+        }
+    }
+
+    /**
+     * Delete data from the filesystem in the order in which the related batches were passed to
+     * {@link #deleteInstances(String, Collection)}.
+     * @param deletionInstance a deletion instance for deleting files
+     */
+    public void deleteFiles(Deletion deletionInstance) {
+        for (final Map.Entry<String, Collection<Long>> deletionBatch : deletionLog) {
+            final ImmutableSetMultimap.Builder<String, Long> targets = ImmutableSetMultimap.builder();
```

Comment:
Could maybe compact this too something like

```
                    ImmutableSetMultimap.<String, Long>builder()
                            .putAll(deletionBatch.getKey(), deletionBatch.getValue())
                            .build()
```

Not essential though, just builder pattern semantics.